content/src/content/jcr_root/apps/core/wcm/components/download/v1/download/clientlibs/editor/js/download.js

```diff
@@ -0,0 +1,96 @@
+/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ ~ Copyright 2018 Adobe Systems Incorporated
+ ~
+ ~ Licensed under the Apache License, Version 2.0 (the "License");
+ ~ you may not use this file except in compliance with the License.
+ ~ You may obtain a copy of the License at
+ ~
+ ~     http://www.apache.org/licenses/LICENSE-2.0
+ ~
+ ~ Unless required by applicable law or agreed to in writing, software
+ ~ distributed under the License is distributed on an "AS IS" BASIS,
+ ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ ~ See the License for the specific language governing permissions and
+ ~ limitations under the License.
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
+/* global jQuery */
+(function($) {
+    "use strict";
+
+    var dialogContentSelector = ".cmp-download__editor";
+    var titleCheckboxSelector = 'coral-checkbox[name="./titleFromAsset"]';
+    var titleTextfieldSelector = 'input[name="./jcr:title"]';
+    var descriptionCheckboxSelector = 'coral-checkbox[name="./descriptionFromAsset"]';
+    var descriptionTextfieldSelector = '.cq-RichText-editable[name="./jcr:description"]';
+    var CheckboxTextfieldTuple = window.CQ.CoreComponents.CheckboxTextfieldTuple.v1;
+    var titleTuple;
+    var descriptionTuple;
+    var fileReference;
+    var $cqFileUpload;
+    var $cqFileUploadEdit;
+
+    $(document).on("dialog-loaded", function(e) {
+        var $dialog = e.dialog;
+        var $dialogContent = $dialog.find(dialogContentSelector);
+        var dialogContent = $dialogContent.length > 0 ? $dialogContent[0] : undefined;
+
+        if (dialogContent) {
+            titleTuple = new CheckboxTextfieldTuple(dialogContent, titleCheckboxSelector, titleTextfieldSelector);
+            descriptionTuple = new CheckboxTextfieldTuple(dialogContent, descriptionCheckboxSelector, descriptionTextfieldSelector, true);
+            $cqFileUpload     = $dialog.find(".cq-FileUpload");
```

Comment:
I'd define constants for all the strings:
".cq-FileUpload", ".cq-FileUpload-edit", "assetselected", "[coral-fileupload-clear]", "cqFileuploadFilereference", "/_jcr_content/metadata.json", "dc:description", "dc:title"